service/src/main/java/uk/gov/hmcts/reform/fpl/service/validators/ChildrenChecker.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.hmcts.reform.fpl.service.validators;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.Child;
+import uk.gov.hmcts.reform.fpl.model.ChildParty;
+
+import java.util.List;
+
+import static org.apache.commons.lang3.ObjectUtils.isEmpty;
+import static uk.gov.hmcts.reform.fpl.service.validators.EventCheckerHelper.allEmpty;
+import static uk.gov.hmcts.reform.fpl.service.validators.EventCheckerHelper.isEmptyAddress;
+import static uk.gov.hmcts.reform.fpl.service.validators.EventCheckerHelper.isEmptyTelephone;
+
+@Component
+public class ChildrenChecker extends PropertiesChecker {
+
+    @Override
+    public List<String> validate(CaseData caseData) {
+        return super.validate(caseData, List.of("children1"));
+    }
+
+    @Override
+    public boolean isStarted(CaseData caseData) {
+        switch (caseData.getAllChildren().size()) {
+            case 0:
+                return false;
+            case 1:
+                return !isEmptyChild(caseData.getAllChildren().get(0).getValue());
+            default:
+                return true;
+        }
+    }
+
+    private static boolean isEmptyChild(Child child) {
+        if (isEmpty(child)) {
+            return true;
+        }
+
+        final ChildParty childParty = child.getParty();
+
+        if (isEmpty(childParty)) {
+            return true;
+        }
+
+        return allEmpty(childParty.getFirstName(),
+                childParty.getLastName(),
+                childParty.getDateOfBirth(),
+                childParty.getGender(),
+                childParty.getLivingSituation(),
+                childParty.getKeyDates(),
+                childParty.getCareAndContactPlan(),
+                childParty.getAdoption(),
+                childParty.getMothersName(),
+                childParty.getFathersName(),
+                childParty.getFathersResponsibility(),
+                childParty.getSocialWorkerName(),
+                childParty.getAdditionalNeeds(),
+                childParty.getShowAddressInConfidentialTab(),
+                childParty.getLitigationIssues())
```

Comment:
I guess we could create a custom validator `isChildEmpty` or something but I guess both ways we can't avoid large statements like this.

Reply:
there were couple options on table, one was to use reflection to get all declared fields and validates if they are empty or null, but there are couple of problems with this. One is that, when user open event which generates e.g default element in collection and there is nested compound element as Address it is not null, it is Address object with all empty fields. To handle this recursive solution should be implemented. Another issue is when object contains List<Elements<>>, elements can be initialised by ccd to have random ids.

One nice option is to created default object instance e.g Child emptyChild = Child.builder.build() and then do emptyChild.equals(actualChild) to check that child is empty, but again it requires equals method to be manually written to handle null, empty, empty Address objects and things like this.

Another problem is that not all fields in object are available to user, and they can be set in backend, but not visible to user. If we user automated solution like reflection, we would detect it as non empty object, but from user pov it is empty

Considered options i choose on purpose to write it this way.